packages/medusa/src/api/middlewares/await-middleware.ts

```diff
@@ -12,7 +12,14 @@ export default (fn: handler): RequestHandler => {
       })
     }
 
-    return fn(req, res).catch(next)
+    try {
+      return await fn(req, res)?.catch(next)
```

Comment:
question/nit(non-blocking): Should be fine to remove the .catch chaining to not mix the approaches? The catch-block should handle both synchronous and asynchronous errors.